paasta_tools/autoscaling/autoscaling_cluster_lib.py

```diff
@@ -123,14 +124,45 @@ def describe_instances(instance_ids, region=None, instance_filters=None):
 
 
 @register_autoscaling_component('aws_spot_fleet_request', CLUSTER_METRICS_PROVIDER_KEY)
-def spotfleet_metrics_provider(spotfleet_request_id, resource, pool_settings):
-    mesos_state = get_mesos_master().state
+def spotfleet_metrics_provider(spotfleet_request_id, resource, pool_settings, config_folder, dry_run=False):
     sfr = get_sfr(spotfleet_request_id, region=resource['region'])
-    if not sfr or not sfr['SpotFleetRequestState'] == 'active':
-        log.error("Ignoring SFR {0} that does not exist or is not active.".format(spotfleet_request_id))
+    if not sfr or sfr['SpotFleetRequestState'] == 'cancelled':
```

Comment:
With the `if not sfr` here, are you worried at all about cleaning up the file just because `get_sfr` returns None for any silly reason? (bad code, api timeout, etc?)